File: app/src/main/java/ru/merkulyevsasha/news/domain/NewsInteractor.java
package ru.merkulyevsasha.news.domain;

import java.util.List;

import io.reactivex.Single;
import ru.merkulyevsasha.news.pojos.Article;

public interface NewsInteractor {
    Single<List<Article>> readNewsAndSaveToDb(int navId);

    Single<Boolean> getFirstRunFlag();

    void setFirstRunFlag();

    Single<List<Article>> selectAll();

    Single<List<Article>> selectNavId(int navId);

    Single<List<Article>> search(String searchTtext);

    boolean needUpdate();
}

File: app/src/main/java/ru/merkulyevsasha/news/domain/NewsInteractor.kt
package ru.merkulyevsasha.news.domain

import io.reactivex.Single
import ru.merkulyevsasha.news.models.Article

interface NewsInteractor {
    fun readArticlesByNavId(navId: Int, searchText: String? = null): Single<List<Article>>
    fun search(searchTtext: String): Single<List<Article>>

    fun refreshArticles(navId: Int): Single<List<Article>>

    fun startRefreshWorker(navId: Int, searchText: String?)
    fun startRefreshWorker()
    fun getProgress(): Single<Boolean>
    fun setProgress(progress: Boolean)
}
